src/abstractdataframe/join.jl

```diff
@@ -241,10 +241,11 @@ Join two `DataFrame` objects
   - `:cross` : a full Cartesian product of the key combinations; every
     row of `df1` is matched with every row of `df2`
 
-* `makeunique` : how to handle columns with duplicate names other than `on` in joined tables:
 
-  - `false` : throw an error if duplicate column names are present
-  - `true` : duplicate column names in `df2` will be deduplicated by adding a suffix
+* `makeunique` : if `false` (the default), an error will be raised
+  if duplicate names are found in columns not joined on;
+  if `true`, duplicate names will be suffixed with `_i`
```

Comment:
Am I right that `i` can only be `1` in this case, since we don't allow duplicated column names in either of the sources?

Reply:
No - it can be a bigger value. See:
```
julia> x = DataFrame(rand(3,3))
3×3 DataFrames.DataFrame
│ Row │ x1       │ x2       │ x3       │
├─────┼──────────┼──────────┼──────────┤
│ 1   │ 0.791145 │ 0.193656 │ 0.238548 │
│ 2   │ 0.756677 │ 0.195111 │ 0.929581 │
│ 3   │ 0.736347 │ 0.234449 │ 0.499932 │

julia> y = DataFrame(rand(3,3))
3×3 DataFrames.DataFrame
│ Row │ x1       │ x2        │ x3       │
├─────┼──────────┼───────────┼──────────┤
│ 1   │ 0.220774 │ 0.0541193 │ 0.735224 │
│ 2   │ 0.147731 │ 0.104057  │ 0.144468 │
│ 3   │ 0.282832 │ 0.0247473 │ 0.188086 │

julia> x[:id] = 1:3
1:3

julia> y[:id] = 1:3
1:3
julia> names!(y, [:x1, :x1, :x1, :id], allow_duplicates=true)
3×4 DataFrames.DataFrame
│ Row │ x1       │ x1_1      │ x1_2     │ id │
├─────┼──────────┼───────────┼──────────┼────┤
│ 1   │ 0.220774 │ 0.0541193 │ 0.735224 │ 1  │
│ 2   │ 0.147731 │ 0.104057  │ 0.144468 │ 2  │
│ 3   │ 0.282832 │ 0.0247473 │ 0.188086 │ 3  │

julia> join(x, y, on=:id)
3×7 DataFrames.DataFrame
│ Row │ x1       │ x2       │ x3       │ id │ x1_3     │ x1_1      │ x1_2     │
├─────┼──────────┼──────────┼──────────┼────┼──────────┼───────────┼──────────┤
│ 1   │ 0.791145 │ 0.193656 │ 0.238548 │ 1  │ 0.220774 │ 0.0541193 │ 0.735224 │
│ 2   │ 0.756677 │ 0.195111 │ 0.929581 │ 2  │ 0.147731 │ 0.104057  │ 0.144468 │
│ 3   │ 0.736347 │ 0.234449 │ 0.499932 │ 3  │ 0.282832 │ 0.0247473 │ 0.188086 │
```

Reply:
Ah, right. It would make sense to have a `left_suffix`/`right_suffix` argument as in R to choose a deduplication suffix to add to columns from the LHS and RHS. We would need to look at what other software does first.

If we improve this before the next release, we could remove `makeunique` without a deprecation.